__tests__/app.test.js

```diff
@@ -5,32 +5,143 @@ const app = require("../app");
 
 beforeEach(() => seed(data));
-// It ensures that the tests start with a predefined dataset or configuration, providing consistent results and avoiding interference between test cases.
 
 afterAll(() => db.end());
-//This ensures that the database connection is properly closed once all the tests have finished running, preventing resource leaks or unnecessary connections.
 
 describe("GET /api/topics", () => {
   test("should return a 200 status code", () => {
     return request(app).get("/api/topics").expect(200);
   });
-  test("should return a 404 status code if path does not excist", () => {
-    return request(app).get("/api/space").expect(404);
-  });
-  //Checks it comes back with an ok code
 
-  //test if path doesnt excist = error test
+  test("should return a 400 status code if path does not exist", () => {
+    return request(app).get("/api/space").expect(400);
+  });
 
   test("should retrieve all topics and return an array of topic objects", () => {
     return request(app)
       .get("/api/topics")
       .expect(200)
       .then((response) => {
-        expect(Array.isArray(response.body)).toBe(true);
-        expect(response.body.length).toBe(3);
-        expect(response.body[0]).toHaveProperty("slug");
-        expect(response.body[0]).toHaveProperty("description");
+        expect(Array.isArray(response.body.topics)).toBe(true);
+        expect(response.body.topics.length).toBe(3);
+
+        response.body.topics.forEach((topic) => {
+          expect(topic).toHaveProperty("slug");
+          expect(topic.slug).toEqual(expect.any(String));
+          expect(topic).toHaveProperty("description");
+          expect(topic.description).toEqual(expect.any(String));
+        });
+      });
+  });
+});
+
+describe("GET /api/", () => {
+  test("should return a 400 status code if path does not exist", () => {
+    return request(app).get("/api/space").expect(400);
+  });
+
+  test("should have a description of all endpoints", () => {
+    const endpointJson = require("../endpoints.json");
+
+    return request(app)
+      .get("/api/")
+      .expect(200)
+      .then((response) => {
+        expect(response.body.api).toEqual(endpointJson);
+      });
+  });
+});
+
+describe("GET /api/articles/:article_id", () => {
+  test("should return a 200 status code", () => {
+    return request(app).get("/api/articles/1").expect(200);
+  });
+
+  test("should return a 404 status code for an article that doesn't exist", () => {
+    const nonExistentArticleId = 76;
+
+    return request(app)
+      .get(`/api/articles/${nonExistentArticleId}`)
+      .expect(404);
+  });
+
+  test("should return the article details for an existing article", () => {
+    const existingArticleId = 6;
+
+    return request(app)
+      .get(`/api/articles/${existingArticleId}`)
+      .expect(200)
+      .then((response) => {
+        expect(Array.isArray(response.articles)).toBe(false);
+        expect(response.body.articles[0]).toHaveProperty("author");
```

Comment:
We can improve this test using a flexible matcher e.g. expect.objectContaining or .toMatchObject.